packages/web3-core/test/unit/web3_middleware_request_manager.test.ts

```diff
@@ -15,130 +15,163 @@ You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { EthExecutionAPI, JsonRpcResponse, Web3APIMethod, Web3APIRequest, Web3APIReturnType } from 'web3-types';
+import {
+	EthExecutionAPI,
+	JsonRpcPayload,
+	JsonRpcRequest,
+	JsonRpcResponse,
+	Web3APIMethod,
+	Web3APIReturnType,
+} from 'web3-types';
 import { jsonRpc } from 'web3-utils';
 import { RequestManagerMiddleware } from '../../src/types';
 import { Web3RequestManager } from '../../src/web3_request_manager';
 
 class Web3Middleware<API> implements RequestManagerMiddleware<API> {
-
-  // eslint-disable-next-line class-methods-use-this
-  public async processRequest<Method extends Web3APIMethod<API>>(
-    request: Web3APIRequest<API, Method>
-  ): Promise<Web3APIRequest<API, Method>> {
-    // Implement the processRequest logic here
-
-    let requestObj = {...request};
-    if (request.method === 'eth_call' && Array.isArray(request.params)) {
-      requestObj = {
-        ...requestObj,
-        params: [...request.params, '0x0', '0x1'],
-      };
-    }
-
-    return Promise.resolve(requestObj);
-  }
-
-  // eslint-disable-next-line class-methods-use-this
-  public async processResponse<
-    Method extends Web3APIMethod<API>,
-    ResponseType = Web3APIReturnType<API, Method>
-  >(
-    response: JsonRpcResponse<ResponseType>
-  ): Promise<JsonRpcResponse<ResponseType>> {
-   
-    let responseObj = {...response};
-    if (!jsonRpc.isBatchResponse(responseObj) && responseObj.id === 1) {
-      responseObj = {
-        ...responseObj,
-        result: '0x6a756e616964' as any,
-      };
-    }
-
-    return Promise.resolve(responseObj);
-  }
+	// eslint-disable-next-line class-methods-use-this
+	public async processRequest<ParamType = unknown[]>(
+		request: JsonRpcPayload<ParamType>,
+	): Promise<JsonRpcPayload<ParamType>> {
+		// Implement the processRequest logic here
+
+		let requestObj = { ...request };
+		if (
+			(requestObj as JsonRpcRequest<ParamType>).method === 'eth_call' &&
+			Array.isArray((requestObj as JsonRpcRequest<ParamType>).params)
+		) {
+			(requestObj as JsonRpcRequest) = {
+				...(requestObj as JsonRpcRequest),
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+				params: [...((requestObj as JsonRpcRequest).params ?? []), '0x0', '0x1'],
```

Comment:
Sorry, what are those fixed numbers `'0x0', '0x1'`?

Reply:
These are numbers for tests only. just add something to the params